packages/components/src/button/src/useDisableClick.tsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * useDisableInteraction provides the original event handler but disables interaction
+ * if the boolean passed is true.
+ */
+import { useCallback, MouseEventHandler } from "react";
+
+export function useDisableClick<T = Element>(disabled: boolean, handleClick: MouseEventHandler<T>) {
+    const handleClickWrapper = useCallback<MouseEventHandler<T>>(
+        event => {
+            if (disabled) {
```

Comment:
Won't this always be `true`?

The function `handleClickWrapper` is only referenceable if `disabled` is true. If the value of `disabled` changes, there will be a stale reference to when `disabled` was true. If `disabled` changes, it won't affect the function until the next rerender, at which point the original `handleClick` will be returned and the new `handleClickWrapper` will be un-referenceable.

Should a `ref` be used instead?

Or just remove the `if` statement and inline the content inside of it?

Reply:
The if looks redundant you're right. I'll remove it in another PR